lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -230,6 +245,9 @@ export const migrateGranulesAndFiles = async (
     granulesResult: granuleMigrationResult,
     filesResult: fileMigrationResult,
   };
+  const filepath = 'granulesAndFilesMigrationErrorLog.json';
```

Comment:
This filename (here and in the executions migration) should probably be timestamped. The race condition you're seeing is probably due to this filename (which is the same throughout all tests) is getting unlinked while it's still being written to by some tests

Reply:
The race condition didn't go away with timestamped names but it occurred significantly less. What I ended up implementing was a function to close the write stream and ensure it finishes [`963d9f7` (#2179)](https://github.com/nasa/cumulus/pull/2179/commits/963d9f7adde0399981f6a5ecdb6b5ee44eca2086#diff-56c5b1137921652643573bffebcd7fae07a919d6517811b49ab0f84759ba10beR29)